node/templates_test.go

```diff
@@ -0,0 +1,92 @@
+package node
+
+import (
+	"testing"
+
+	"github.com/NebulousLabs/Sia/siatest"
+)
+
+// TestNewNode is a basic smoke test for NewNode that uses all of the templates
+// to verify a working NewNode function.
+func TestNewNode(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+
+	// Test AllModulesTemplate.
+	dir, err := siatest.TestDir("node", t.Name()+"-AllModulesTemplate")
+	if err != nil {
+		t.Fatal(err)
+	}
+	n, err := NewNode(AllModules(dir))
```

Comment:
creating a node is pretty easy. If you already have the dir, just call `NewNode(AllModules(dir))`